content/documentation/configuration.md

```diff
@@ -68,34 +71,64 @@ The providers configuration tells Tegola where your data lives. Data providers e
-| user                | Yes      |          | The database user                                  |
-| password            | Yes      |          | The database user's password                       |
+| database            | No       |          | The name of the database                           |
+| user                | No       |          | The database user                                  |
+| password            | No       |          | The database user's password                       |
 | srid                | No       | 3857     | The default SRID for this data provider            |
+| ssl_mode            | No       | prefer   | PostGIS SSL mode.                                  |
 | max_connections     | No       | 100      | max number of connections in the connection pool.  |
+| max_connection_idle_time     | No       | 30m      | max connection idle time.  |
+| max_connection_lifetime     | No       | 1h      | max connection lifetime.  |
+
+Establishing a connection via connection string (`uri`) will become the default connection method as of v0.16.0.
+Connecting via host/port/database is flagged for deprecation as of v0.15.0 but will be possible until v0.16.0 still.
+
+**Example**
+
+```
+# {protocol}://{user}:{password}@{host}:{port}/{database}?{options}=
+
+postgres://tegola:supersecret@localhost:5432/tegola?sslmode=prefer&pool_max_conns=10
+```
+
+**Options**
+
+- `sslmode`: [Optional] PostGIS SSL mode. Default: "prefer"
+- `pool_max_conns`: [Optional] The max connections to maintain in the connection pool. Defaults to 100. 0 means no max.
+- `pool_max_conn_idle_time`: [Optional] The maximum time an idle connection is kept alive. Defaults to "30m".
+- `max_connection_lifetime` [Optional] The maximum time a connection lives before it is terminated and recreated. Defaults to "1h".
 
 **Example PostGIS Provider config**
 
 ```toml
 [[providers]]
-name = "test_postgis"   # provider name is referenced from map layers
-type = "postgis"        # the type of data provider.
-host = "localhost"      # postgis database host
-port = 5432             # postgis database port
-database = "tegola"     # postgis database name
-user = "tegola"         # postgis database user
-password = ""           # postgis database password
+name = "test_postgis"       # provider name is referenced from map layers (required)
+type = "postgis"            # the type of data provider must be "postgis" for this data provider (required)
+
+uri = "postgres://tegola:supersecret@localhost:5432/tegola?sslmode=prefer" # PostGIS connection string (required)
+
+host = "localhost"                # PostGIS database host (deprecated)
+port = 5432                       # PostGIS database port (deprecated)
+database = "tegola"               # PostGIS database name (deprecated)
+user = "tegola"                   # PostGIS database user (deprecated)
+password = "supersecret"          # PostGIS database password (deprecated)
+ssl_mode = "prefer"
```

Comment:
I think we should removed these from the example if `uri` is present as it's a bit confusing to show both in the config block.

Reply:
fair enough. deleted in all options and examples in favour of the uri